internal/internal_task_handlers.go

```diff
@@ -75,15 +75,17 @@ type (
 	// workflowTask wraps a decision task.
 	workflowTask struct {
 		task            *s.PollForDecisionTaskResponse
+		autoConfigHint  *s.AutoConfigHint
```

Comment:
why can not we reuse the autoConfigHint inside task(PollForDecisionTaskResponse) here? I'm assuming task != nil will mess up the logic somewhere else

Reply:
I haven't tried. But the field is widely used in many different places. Breaking the Nilness assumption might mess it up.

Reply:
I don't think we need the new field. It's a redundant information and requires each instantiation of workflowTask/activityTask to fill it based on the response. 
Instead expose a helper function such as:
```
func (wt *workflowTask) GetAutoConfigHint() *s.AutoConfigHint {
   if wt.task == nil {
     return nil
  }
  return wt.task.GetAutoConfigHint()
```

Less code to write and maintain.

Reply:
yeah, I checked carefully. We can populate this task field for empty polls. Thus I've changed the title as well.